### PR TITLE
add shellcheck workflow

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

#### Summary and Scope

- Partially implements MTL-1497

##### Issue Type

- RFE Pull Request

Adds shellcheck workflow to this repo.

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (x) (if yes, please include results or a description of the test)
 
#### Idempotency
 

 
N/A

#### Risks and Mitigations
 

Low

### CODE REVIEW

File: bin/bios-baseline.sh [hunk 1]
```diff
@@ -220,12 +220,12 @@ function run_ilo() {
             ilorest --nologo login ${ncn_bmc} -u ${bmc_username} -p ${bmc_password} >/dev/null
         fi
 
-        ilo_config 2>&1 >$LOG_DIR/${ncn_bmc}.log
-        ilorest --nologo logout 2>&1 >/dev/null
+        ilo_config >$LOG_DIR/${ncn_bmc}.log 2>&12>&1
```
Comment (heemstra) [inline bin/bios-baseline.sh:223]: `&12` you say?

Comment (jacobsalmela) [reply bin/bios-baseline.sh:223]: Oops.  I'll fix that up

Comment (jacobsalmela) [reply bin/bios-baseline.sh]: ok I meant to just to the shellcheck disable for these.  I committed my attempted changes, but everything is fixed up now to just disable these warnings for now

**Review by rustydb — CHANGES_REQUESTED** [review]
let's fix this PR first, there's enough bash in here but small enough scripts to warrant it.

File: bin/bios-baseline.sh [hunk 2]
```diff
@@ -1,27 +1,27 @@
 #!/bin/bash
```
Comment (rustydb) [inline bin/bios-baseline.sh:1]: This file has too many to fix, and the ilorest stuff is _really_ temperamental.. we can pass this one without fixing it up.

**Review by rustydb — APPROVED** [review]

**Pull request merged**